### PR TITLE
Enable scoped parameters in signatures of source-generated interop methods

### LINKED ISSUE #77624: LibraryImportGenerator does not respect C# 11 `scoped` modifier on parameters
### Description

A `scoped` modifier applied to a P/Invoke attributed with `[LibraryImport(...)]` is not replicated to the generated marshaling method.

### Reproduction Steps

```csharp
[LibraryImport("c", SetLastError = true)]
public static partial nint read(int fildes, scoped Span<byte> buf, nuint nbyte);
```

### Expected behavior

```csharp
[System.CodeDom.Compiler.GeneratedCodeAttribute("Microsoft.Interop.LibraryImportGenerator", "7.0.6.47203")]
public static partial nint read(int fildes, scoped global::System.Span<byte> buf, nuint nbyte)
{
    // ...
}
```

### Actual behavior

```csharp
[System.CodeDom.Compiler.GeneratedCodeAttribute("Microsoft.Interop.LibraryImportGenerator", "7.0.6.47203")]
public static partial nint read(int fildes, global::System.Span<byte> buf, nuint nbyte)
{
    // ...
}
```

`error CS8988: The 'scoped' modifier of parameter 'buf' doesn't match partial method declaration.`

### Regression?

_No response_

### Known Workarounds

_No response_

### Configuration

* .NET 7 RC 2
* Windows 11
* x64

### Other information

_No response_

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Fixes #77624

### CODE REVIEW

File: src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/SignatureContext.cs [hunk 1]
```diff
@@ -43,9 +42,24 @@ public IEnumerable<ParameterSyntax> StubParameters
...
+                        // All other cases of explicit parameters are only scoped when the "scoped" keyword is present.
+                        // When the "scoped" keyword is present, it must be present on all declarations.
+                        if (typeInfo.ScopedKind != ScopedKind.None && typeInfo.RefKind != RefKind.Out)
+                        {
+                            tokens = tokens.Add(Token(SyntaxKind.ScopedKeyword));
+                        }
+
+                        if (typeInfo.IsByRef)
+                        {
+                            tokens = tokens.Add(Token(typeInfo.RefKindSyntax));
```
Comment (AaronRobinsonMSFT) [inline src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/SignatureContext.cs:57]: We didn't conditionally add this previously, was that a bug?

Comment (jkoritzinsky) [reply src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/SignatureContext.cs:57]: Technically we were adding a token for `SyntaxKind.None`. That ends up being an empty string when we stringify everything, so it didn't do anything wrong, but it's not necessary (and we get marginally better formatting by not including it when not needed)

Comment (AaronRobinsonMSFT) [reply src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/SignatureContext.cs:57]: Oh, we were just propagating details from the current declaration. This make sense.

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Comment by jkoritzinsky** [comment]
Test failures are unrelated.

**Pull request merged**